src/xai_grok/errors/unauthorized_user_error.py

```diff
@@ -0,0 +1,8 @@
+import abc
+
+
+class UnauthorizedUserError(Exception, abc.ABC):
+
+    @abc.abstractmethod
+    def __init__(self, message: str):
+        super().__init__(message)
```

Comment:
The `UnauthorizedUserError` class is abstract due to the `@abc.abstractmethod` decorator on `__init__`, making it impossible to instantiate. Remove the `@abc.abstractmethod` decorator to allow instantiation.